src/jdk.incubator.vector/share/classes/jdk/incubator/vector/VectorMathLibrary.java

```diff
@@ -0,0 +1,334 @@
+     *               We use "u10" for all operations by default
+     *               But for those functions do not have u10 support, we use "u05" instead
+     *     <suffix> indicates neon/sve/rvv
+     *              "sve/advsimd/rvv" for sve/neon/rvv implementations
+     *     e.g. sinfx_u10sve is the method for computing vector float sin using SVE instructions
+     *          cosd2_u10advsimd is the method for computing 2 elements vector double cos using NEON instructions
+     */
+    private static class SLEEF implements Library {
+        static {
+            VectorSupport.loadNativeLibrary("sleef");
+        }
+
+        private static String suffix(VectorShape vshape) {
+            if (isAARCH64()) {
+                if (vshape.vectorBitSize() <= 128) {
+                    return "advsimd";
+                } else {
+                    assert (vshape == VectorShape.S_Max_BIT) : "not supported: " + vshape;
+                    return "sve";
+                }
+            } else if (isRISCV64()) {
+                assert (vshape == VectorShape.S_Max_BIT) : "not supported: " + vshape;
+                return "rvv";
+            } else {
+                throw new InternalError("unsupported platform");
+            }
+        }
+
+        private static String precisionLevel(Operator op) {
+            return (op == HYPOT ? "u05" : "u10");
+        }
+
+        @Override
+        public String symbolName(Operator op, VectorSpecies<?> vspecies) {
+            int vlen = (vspecies == FloatVector.SPECIES_64 ? 4 : vspecies.length()); // reuse 128-bit variant for 64-bit float vectors
+            boolean isShapeAgnostic = isRISCV64() || (isAARCH64() && vspecies.vectorBitSize() > 128);
+            return String.format("%s%s%s_%s%s", op.operatorName(),
+                                 (vspecies.elementType() == float.class ? "f" : "d"),
+                                 (isShapeAgnostic ? "x" : Integer.toString(vlen)),
+                                 precisionLevel(op),
+                                 suffix(vspecies.vectorShape()));
+        }
+
+        @Override
+        public boolean isSupported(Operator op, VectorSpecies<?> vspecies) {
+            Class<?> etype = vspecies.elementType();
+            if (etype != float.class && etype != double.class) {
+                return false; // only FP element types are supported
+            }
+            int maxLaneCount = VectorSupport.getMaxLaneCount(vspecies.elementType());
+            if (vspecies.length() > maxLaneCount) {
+                return false; // lacking vector support
+            }
+            if (vspecies.vectorShape() != VectorShape.S_Max_BIT) {
+                if (isRISCV64()) {
+                    return false; // FIXME: only MAX shapes are supported on RISC-V
+                }
+                if (isAARCH64() && vspecies.vectorBitSize() > 128) {
+                    return false; // FIXME: SVE support only for MAX shapes
+                }
```

Comment:
SVE also supports operations for partial vector size, which means the vector size is smaller than MAX size. For example, if the max vector size of a SVE architecture is 512-bit, the FP vector operations with `VectorShape.S_256_BIT` are also supported. They are implemented with the same scalable math functions in SLEEF.

Hence, I think this check and the assertion in line-198 can be removed. Thanks!

Reply:
Please see the `addr` definition code in https://github.com/openjdk/jdk/blob/master/src/hotspot/share/opto/vectorIntrinsics.cpp#L1877 . If queried `addr` returns `nullptr` for 256-bit vectors, and the arch supports scalable vector, then the `addr` will be assigned to the scalable ones.

Reply:
Ah, ok. Thanks for the pointer. I aligned lookup logic with existing behavior. 

I'd like to double-check one things: is it fine to use scalable vector variants for fixed-sized vector shapes of smaller size without any explicit masking/stripping of upper vector part?

Reply:
Yes, it's fine for lanewise oeprations. For others like cross-lanes, stores and other operations, we will generate a predicate to strip the upper vector part. The op list that need the masking are listed here: https://github.com/openjdk/jdk/blob/master/src/hotspot/cpu/aarch64/aarch64_vector.ad#L280